utilities/pipelines/resourceRemoval/helper/Invoke-ResourceRemoval.ps1

```diff
@@ -135,6 +135,21 @@ function Invoke-ResourceRemoval {
             }
             break
         }
+        'Microsoft.DevTestLab/labs' {
+            $resourceGroupName = $resourceId.Split('/')[4]
+            $resourceName = Split-Path $resourceId -Leaf
+            if ($PSCmdlet.ShouldProcess("DevTestLab Lab [$resourceName]", 'Remove')) {
+                Get-AzResourceLock -ResourceGroupName $resourceGroupName |
+                    Where-Object -FilterScript { $PSItem.properties.notes -eq "Reserved resource locked by '$resourceName' lab." } |
+                    ForEach-Object {
+                        $null = Remove-AzResourceLock -LockId $PSItem.LockId -Force
+                        Write-Verbose "Removed lock [$($PSItem.Name)] created by the DevTest Lab [$resourceName] on resource [$($PSItem.ResourceName)]. Waiting 10 seconds for propagation." -Verbose
+                        Start-Sleep 10
+                    }
+                $null = Remove-AzResource -ResourceId $resourceId -Force -ErrorAction 'Stop'
```

Comment:
I'm a bit confused why we have this custom logic if the removal is a 'normal' removal in the end - unless - the lock removal is something very special to the DevTestLabs. If it turns out the default propagation time is too short, we should simply update it there in my opinion.

Reply:
This is a special lock implemented by the DTL service, it's outside our templates deployment scope When running the normal removal process, hence it's not something that gets removed. Another example here is that DTL deploys its own KV and Storage account that only get removed when DTL is removed. It automatically removes the locks for these services but once you want to use an existing storage account instead of the created DTL one, the lock stays behind.. unfortunately it's not something we can control